lib/puppet/type/elasticsearch_plugin.rb

```diff
@@ -0,0 +1,31 @@
+Puppet::Type.newtype(:elasticsearch_plugin) do
+
+  @doc = "Plugin installation type"
+  
+  ensurable do
+    defaultvalues
+    defaultto :present
+  end
+
+  newparam(:name, :namevar => true) do
+    desc 'An arbitrary name used as the identity of the resource.'
+  end
+
+  newparam(:url) do
+    desc 'Url of the package'
+  end
+
+  newparam(:source) do
+    desc 'Source of the package. puppet:// or file:// resource'
+  end
+
+  newparam(:proxy_args) do
+    desc 'Proxy Host'
```

Comment:
Possibly, but the current define already supports it so wanted to keep the same support in there.